### DISCUSSION TITLE
Migrate from QS Reremake mySQL database to QS Hikari mySQL database

### DISCUSSION THREAD

**Discussion by Bloodlex** [QuickShop-Community/QuickShop-Hikari#623] [Q&A] [answered]
Hello 🙂,

I would like to ask if you could give me a step by step instruction how to migrate from QS Reremake with mySQL database to QS Hikari with mySQL database.

I saw the instructions for a file storage, but I'm not sure how to proceed with already existing database.

Thank you very much.

Best regards,
Bloodlex

**Comment by Ghost-chu** [comment]
In fact, I do not highly recommend using the Migrator. If possible, you can just migrate the store database and make changes directly from Hikari's configuration file.

Hikari's migration server is quite old, I'm not sure if it still working with Reremake.

If you plan to perform the migration on the production server, I suggest that we negotiate a time, and I will assist you in the migration on Discard.

In this way, if there are problems in the migration process, we can solve them in time and immediately compile the new build to continue the migration.

In next 7 days, I have free time at 17:00-23:50 (BJT).

**Comment by Ghost-chu** [comment, marked as answer]
UPDATE: Let me update this discussion: Hikari now have official ways to migrate from Reremake to Hikari: https://quickshop-community.github.io/QuickShop-Hikari-Documents/docs/addon/reremake-migrator

---
***The following is outdated,Please check the link first***  

If you need self migrating, here is steps:

1. Execute `/qs export` to export your QuickShop-Reremake database as JSON.
2. Stop your server, backup EVERYTHING (include worlds, playerdatas, plugins, plugin files, databases).
3. Remove QuickShop-Reremake's jar from your server, upload QuickShop-Hikari's jar.
4. Rename `QuickShop` directory to `QuickShop-Hikari`, keep all files inside, don't touch them.
5. Fire up your server, waiting QuickShop-Hikari resolving the dependencies, self-testing and check mirating availablity.
6. If all checks passed, you need follow to create a `continue.txt` in QuickShop data folder, then converting will start.
7. After migrating finished, stop your server, open `config.yml`, turn on legacy updater:

```yaml
legacy-updater:
  shop-sign: false # set this to true
```

8. Re-configure your `interaction.yml` and `price-restriction.yml`.
9. If possible, you can delete your `config.yml`, `group.yml`, `interaction.yml`, `items-lookup.yml`, `price-restriction.yml`, let plugin regenerate them to start up in fresh configuration to avoid strange issues (don't forget re-enable `legacy-updater`, otherwise all Reremake info signs will be unrecognized.

**Comment by Ghost-chu** [comment]
Please note that Hikari's data structure and storage method are very different from those of Remake, so if there is no backup, it cannot be rolled back.

**Reply by Bloodlex** [reply]
Hey 🙂 

Thank you very much, you are very kind 🙂 I'll do a full backup and try to do it myself following your instructions, but if something would break, I'll be glad if you could assist me in that. I really appreciate, that you are always so nice and helpful 🙂.

I will get back to you with details after the migration, hopefully everything will go as planned 🙂.

Best regards

**Comment by Ghost-chu** [comment]
Let me update this discussion: Hikari now have official ways to migrate from Reremake to Hikari: https://quickshop-community.github.io/QuickShop-Hikari-Documents/docs/addon/reremake-migrator
